lib/src/models/interaction.dart

```diff
@@ -275,3 +274,86 @@ class MultiselectInteraction extends ComponentInteraction implements IMultiselec
     values = (raw["data"]["values"] as List<dynamic>).cast<String>();
   }
 }
+
+abstract class IResolvedSelectInteraction implements IComponentInteraction {
+  /// Iterable of all ids selectionned.
+  Iterable<Snowflake> get values;
+}
+
+abstract class ResolvedSelectInteraction extends ComponentInteraction implements IResolvedSelectInteraction {
+  @override
+  late final Iterable<Snowflake> values;
+
+  ResolvedSelectInteraction(INyxx client, RawApiMap raw) : super(client, raw) {
+    values = (raw['data']['values'] as List).cast<String>().map(Snowflake.new);
+  }
+}
+
+abstract class IUserMultiSelectInteraction implements IResolvedSelectInteraction {
+  /// The users that were selected.
+  Iterable<IUser> get users;
+
+  /// The [IMember]s attached to the [users].
+  Iterable<IMember> get members;
+}
+
+class UserMultiSelectInteraction extends ResolvedSelectInteraction implements IUserMultiSelectInteraction {
+  @override
+  late final Iterable<IUser> users;
+  @override
+  late final Iterable<IMember> members;
+
+  UserMultiSelectInteraction(INyxx client, RawApiMap raw) : super(client, raw) {
+    final resolved = InteractionDataResolved(raw['data']['resolved'] as RawApiMap, guild?.id, client);
+    users = resolved.users;
+    members = resolved.members;
+  }
+}
+
+abstract class IRoleMultiSelectInteraction implements IResolvedSelectInteraction {
+  /// The roles that were selected.
+  Iterable<IRole> get roles;
+}
+
+class RoleMultiSelectInteraction extends ResolvedSelectInteraction implements IRoleMultiSelectInteraction {
+  @override
+  late final Iterable<IRole> roles;
+
+  RoleMultiSelectInteraction(INyxx client, RawApiMap raw) : super(client, raw) {
+    final resolved = InteractionDataResolved(raw['data']['resolved'] as RawApiMap, guild?.id, client);
+    roles = resolved.roles;
+  }
+}
+
+abstract class IMentionableMultiSelectInteraction implements IResolvedSelectInteraction {
+  /// The mentionables that were selected.
+  Iterable<Mentionable> get mentionables;
+}
+
+class MentionableMultiSelectInteraction extends ResolvedSelectInteraction implements IMentionableMultiSelectInteraction {
+  @override
+  late final Iterable<Mentionable> mentionables;
+
+  MentionableMultiSelectInteraction(INyxx client, RawApiMap raw) : super(client, raw) {
+    final resolved = InteractionDataResolved(raw['data']['resolved'] as RawApiMap, guild?.id, client);
+    mentionables = [
+      ...{...resolved.users, ...resolved.members},
+      ...resolved.roles
```

Comment:
coudnt we just do `...resolved.users, ...resolved.members, ...resolved.roles` ?

Reply:
Yeah